lib/AFS/CellCC/Dump.pm

```diff
@@ -41,36 +41,51 @@ our @EXPORT_OK = qw(process_dumps);
 # currently just calculate a checksum for the dump blob, and update the
 # database to give the dump filename and metadata.
 sub
-_dump_success($$$$) {
-    my ($jobid, $dvref, $prev_state, $dump_fh) = @_;
+_dump_success($$$) {
+    my ($prev_state, $jobs_ref, $dump_fhs_ref) = @_;
+    my @jobs = @{$jobs_ref};
+    my @dump_fhs = @{$dump_fhs_ref};
     my $done_state = 'DUMP_DONE';
-    my $filesize = stat($dump_fh)->size;
+    my $filesize = stat($dump_fhs[0])->size;
 
-    my $base_file = basename($dump_fh->filename);
-
-    seek($dump_fh, 0, 0);
+    seek($dump_fhs[0], 0, 0);
 
+    for my $i (1 .. $#dump_fhs) {
+        if(-e $dump_fhs[$i]->filename) {
+            unlink($dump_fhs[$i]->filename)
```

Comment:
just a note here: I was a little concerned that unlinking he files out from under File::Temp's files would do something weird (since we separately have a filehandle for the same file). It seems to be fine: File::Temp does have functionality to "safely delete" the underlying file (checking if it still points to the same file as we started with), but the default behavior doesn't really check for that.

I'm not complaining about it (I don't see any better way of doing this), but it's just worth being aware of that this is a little weird.